test/core/end2end/fuzzers/api_fuzzer.cc

```diff
@@ -41,10 +44,28 @@
 #include "src/libfuzzer/libfuzzer_macro.h"
 #include "test/core/end2end/data/ssl_test_data.h"
 #include "test/core/end2end/fuzzers/api_fuzzer.pb.h"
+#include "test/core/event_engine/fuzzing_event_engine/fuzzing_event_engine.h"
 #include "test/core/util/passthru_endpoint.h"
 
-static constexpr uint64_t kMaxAdvanceTimeMicros =
-    31536000000000;  // 1 year (24 * 365 * 3600 * 1000000)
+static grpc_event_engine::experimental::FuzzingEventEngine* g_event_engine =
+    new grpc_event_engine::experimental::FuzzingEventEngine(
+        grpc_event_engine::experimental::FuzzingEventEngine::Options());
+static int g_unused_initialize_event_engine = []() {
+  grpc_event_engine::experimental::SetDefaultEventEngineFactory(
+      new std::function<std::unique_ptr<
+          grpc_event_engine::experimental::EventEngine>()>([]() {
+        // HACK HACK HACK
+        // We know that this event engine will never be deleted by the
+        // caller, instead release() will be called and the value stashed in
```

Comment:
If the engine can be put into a bad state, that's going to hurt subsequent test runs. Why not create a new engine each time?

Reply:
I don't think the engine can be put into a bad state?

I'd like to create a new engine each fuzzer run, but it'll require a change to how we track the default event engine (that's not resettable), and I think we can deal with that as a separate work item.